python/lsst/afw/table/catalogMatches.py

```diff
@@ -254,3 +256,112 @@ def copyAliasMapWithPrefix(inSchema, outSchema, prefix=""):
         outSchema.getAliasMap().set(prefix + k, prefix + v)
 
     return outSchema
+
+
+def reindexCatalog(catalog, indices, deep=True):
+    """Apply a numpy index array to an afw Catalog
+
+    Parameters
+    ----------
+    catalog : subclass of `lsst.afw.table.BaseCatalog`
+        Catalog to reindex.
+    indices : `numpy.ndarray` of `int`
+        Index array.
+
+    Returns
+    -------
+    new : subclass of `lsst.afw.table.BaseCatalog`
+        Reindexed catalog. Records are shallow copies of those in ``catalog``.
+    """
+    new = type(catalog)(catalog.schema)
```

Comment:
I am not confident this is a reliable way to construct a new catalog that contains everything you might want from the previous one.

Or, rather, this one definitely doesn't, and it's not appropriate for shallow-copies - `type(catalog)(catalog.table)` would be closer, but I think the safe thing to do is to just say `SourceCatalog(catalog.table)`, which is definitely safe, and document this as something that doesn't work for other catalog types.

Reply:
That's a shame, but I'm fine with changing it to `SourceCatalog`. If someone does need to reindex a different type of catalog at least this will provide a starting point.